network/stream/messages.go

```diff
@@ -225,9 +225,14 @@ func (p *Peer) handleOfferedHashesMsg(ctx context.Context, req *OfferedHashesMsg
 	for i := 0; i < lenHashes; i += HashSize {
 		hash := hashes[i : i+HashSize]
 
-		if wait := c.NeedData(ctx, hash); wait != nil {
+		log.Trace("checking offered hash", "ref", fmt.Sprintf("%x", hash))
+
+		if _, wait := c.NeedData(ctx, hash); wait != nil {
```

Comment:
so we are still asking all offering peers, not just the first. note TODO here

Reply:
Yes, we are still asking all peers. We decided that we will be addressing this later, in order to reduce the scope of an already big PR.